alerting/src/main/kotlin/org/opensearch/alerting/actionconverter/DestinationActionsConverter.kt

```diff
@@ -0,0 +1,392 @@
+import org.opensearch.commons.notifications.action.GetNotificationConfigRequest
+import org.opensearch.commons.notifications.action.GetNotificationConfigResponse
+import org.opensearch.commons.notifications.action.UpdateNotificationConfigRequest
+import org.opensearch.commons.notifications.model.Chime
+import org.opensearch.commons.notifications.model.ConfigType
+import org.opensearch.commons.notifications.model.Email
+import org.opensearch.commons.notifications.model.Feature
+import org.opensearch.commons.notifications.model.NotificationConfig
+import org.opensearch.commons.notifications.model.NotificationConfigInfo
+import org.opensearch.commons.notifications.model.Slack
+import org.opensearch.commons.notifications.model.Webhook
+import org.opensearch.index.Index
+import org.opensearch.index.shard.ShardId
+import org.opensearch.rest.RestStatus
+import org.opensearch.search.sort.SortOrder
+import java.net.URI
+import java.net.URISyntaxException
+import java.util.EnumSet
+
+class DestinationActionsConverter {
+
+    companion object {
+        private val logger = LogManager.getLogger(DestinationActionsConverter::class)
+
+        private val INVALID_DESTINATION_TYPES = listOf(ConfigType.NONE, ConfigType.EMAIL_GROUP, ConfigType.SMTP_ACCOUNT)
+        private val ALL_DESTINATION_CONFIG_TYPES = ConfigType.values().filter { !INVALID_DESTINATION_TYPES.contains(it) }
+
+        fun convertGetDestinationsRequestToGetNotificationConfigRequest(request: GetDestinationsRequest): GetNotificationConfigRequest {
+            val configIds: Set<String> = if (request.destinationId != null) setOf(request.destinationId) else emptySet()
+            val table = request.table
+            val fromIndex = table.startIndex
+            val maxItems = table.size
+            val sortOrder: SortOrder = SortOrder.fromString(table.sortOrder)
+            val filterParams = HashMap<String, String>()
+            if (request.destinationType == "ALL") {
+                filterParams["config_type"] = ALL_DESTINATION_CONFIG_TYPES.joinToString(",")
+            } else {
+                if (request.destinationType == DestinationType.CUSTOM_WEBHOOK.value) {
+                    filterParams["config_type"] = ConfigType.WEBHOOK.tag
+                } else {
+                    filterParams["config_type"] = request.destinationType
+                }
+            }
+            if (!table.searchString.isNullOrBlank()) {
+                filterParams["name"] = table.searchString
+            }
+
+            val sortField = when (table.sortString) {
+                "destination.name.keyword" -> NAME_TAG
+                "destination.type" -> CONFIG_TYPE_TAG
+                "destination.last_update_time" -> UPDATED_TIME_TAG
+                else -> null
+            }
+
+            return GetNotificationConfigRequest(configIds, fromIndex, maxItems, sortField, sortOrder, filterParams)
+        }
+
+        fun convertGetNotificationConfigResponseToGetDestinationsResponse(
```

Comment:
Can we mark this as internal as well, Since it is also being used inside another function?